### ISSUE TITLE
[Bug]: [date 4.1]tke regression: load data mo reported error SIGSEGV: segmentation violation

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#15288]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

6920986641050527e79339993501e57a3e4b813a

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

job:https://github.com/matrixorigin/mo-nightly-regression/actions/runs/8509358350/job/23306720538
<img width="1621" alt="企业微信截图_a95c782a-e2c3-4929-ac17-8ed221eacad3" src="https://github.com/matrixorigin/matrixone/assets/113406637/9f65f791-aeea-4a1d-8cff-45281b22f85e">

回归环境3cn和1dn都有重启现象
<img width="1047" alt="企业微信截图_63e21990-3fc2-4116-8a03-45057312b547" src="https://github.com/matrixorigin/matrixone/assets/113406637/ed6716fd-1752-47bd-8260-4cd2e4c2b063">

日志报SIGSEGV: segmentation violation
<img width="1769" alt="企业微信截图_a1419332-152d-4ff5-811d-d788a60cdf40" src="https://github.com/matrixorigin/matrixone/assets/113406637/3a4ccea3-caec-4370-8947-dd975009818a">
https://grafana.ci.matrixorigin.cn/explore?panes=%7B%22s4G%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22mo-nightly-regression-20240401%5C%22%7D%20%7C%3D%20%60%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221711986808583%22,%22to%22:%221711986979583%22%7D%7D%7D&schemaVersion=1&orgId=1

main最新的commit：53852bde24932c319b2b2d0f5cc5e7ee1fa56bfc，也是必现该错误
<img width="1516" alt="企业微信截图_bbd8b166-f6a6-4c44-9fb1-8a931317c390" src="https://github.com/matrixorigin/matrixone/assets/113406637/28c88249-6090-429c-bc27-5e9ea0007a03">
根据二分，怀疑该pr导致，麻烦定位@gouhongshen
<img width="1417" alt="企业微信截图_b1e45665-2617-41f2-948b-d5b1b1d52191" src="https://github.com/matrixorigin/matrixone/assets/113406637/af958736-0e01-4de9-8b0d-37339291c634">

### Steps to Reproduce

```Markdown
tke regression tpcc test
```

**Comment by gouhongshen** [comment]
reproduced after the revert merged:

job: https://github.com/matrixorigin/mo-nightly-regression/actions/runs/8524583157/job/23360705919

log: 
https://grafana.ci.matrixorigin.cn/explore?panes=%7B%22utQ%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22mo-nightly-regression-20240402%5C%22%7D%20%7C%3D%20%60%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221712085115765%22,%22to%22:%221712085595765%22%7D%7D%7D&schemaVersion=1&orgId=1

**Comment by gouhongshen** [comment]
core dump 文件地址：nightly-devops: /data/tmp/core/core-dump
对应的 cn: nightly-regression-dis-tp-cn-w7shs-mo-nightly-regression-20240402
对应的 log: 上一条 comment
对应的 job: 上一条 comment
对应的 commit id: 47b7ee82f41fa63573e8025bc947e9d7c479129b

**Comment by gouhongshen** [comment]
处理其他 s-1

**Comment by gouhongshen** [comment]
定位到第三方 hash 库的 hash 过程，该 hash 过程被编译优化成了汇编代码，可能与平台的兼容性有关系。
现在正在尝试两种方式：
1. 替换 hash 函数
2. 禁止优化成汇编代码

**Comment by XuPeng-SH** [comment]
should be fixed by #15418

**Comment by heni02** [comment]
confirm，closed
commit：123d3af09
job：https://github.com/matrixorigin/mo-nightly-regression/actions/runs/8618339949/job/23620887110
<img width="1389" alt="企业微信截图_085e0ca5-0537-47f0-a211-222727259075" src="https://github.com/matrixorigin/matrixone/assets/113406637/4cefc5f2-3f6b-42f1-a400-b16a75d59a5b">
